File: 3658_GCD_of_Odd_and_Even_Sums.c
#include <stdio.h>

int gcdOfOddEvenSums(int n) {
    int sumOdd = 0, sumEven = 0;

    for(int i = 1; i <= n*2; i++){
        if(i % 2 == 0)
            sumEven += i;
        else
            sumOdd += i;
    }

    int temp;
    while (sumOdd != 0) {
        temp = sumOdd;
        sumOdd = sumEven % sumOdd;
        sumEven = temp;
    }

    return sumEven;
}

int main() {
    int n;
    printf("Enter a positive integer n: ");
    scanf("%d", &n);
    
    int result = gcdOfOddEvenSums(n);
    printf("GCD of sums of odd and even integers from 1 to %d is: %d\n", n * 2, result);
    
    return 0;
}

File: 3658_GCD_of_Odd_and_Even_Sums.py
import math
class Solution:
    def gcdOfOddEvenSums(self, n: int) -> int:
        sumOdd, sumEven = 0, 0
        for num in range(1,n*2+1):
            if num % 2 == 0:
                sumEven += num
            else:
                sumOdd += num

        return math.gcd(sumOdd, sumEven)
            
a = Solution()
print(a.gcdOfOddEvenSums(4))  # Output: 4
print(a.gcdOfOddEvenSums(5))  # Output: 5
print(a.gcdOfOddEvenSums(1009))  # Output: 1